.github/workflows/test.yml

```diff
@@ -0,0 +1,138 @@
+name: CI
+
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron: "0 0 * * 3"
+  workflow_dispatch:
+
+jobs:
+  resources:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Create resource cache
+        id: cache
+        uses: actions/cache@v3
+        with:
+          path: ./fortran_tests/before/*/
+          key: resources-${{ github.event_name }}
+
+      - name: Prepare tests (default)
+        if: ${{ steps.cache.outputs.cache-hit != 'true' }}
+        run: |
+          .travis/prep_regular.sh
+
+      - name: Prepare tests (schedule)
+        if: ${{ steps.cache.outputs.cache-hit != 'true' && github.event_name == 'schedule' }}
+        run: |
+          .travis/prep_cron.sh
+
+  conda:
+    needs:
+      - resources
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        python: ["3.7", "3.8", "3.9", "3.10"]
+
+    defaults:
+      run:
+        shell: "bash -l {0}"
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Load resources
+        uses: actions/cache@v3
+        with:
+          path: ./fortran_tests/before/*/
+          key: resources-${{ github.event_name }}
+
+      - name: Install dependencies
+        uses: mamba-org/provision-with-micromamba@main
+        with:
+          environment-file: environment.yml
+          extra-specs: |
+            python=${{ matrix.python }}
+            coveralls
+
+      - name: Install project
+        run: pip install .
+
+      - name: Run tests
+        run: |
+          coverage run --source=fprettify setup.py test
+
+      - name: Coverage upload
+        run: coveralls --service=github
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          COVERALLS_FLAG_NAME: ${{ matrix.python }}
+          COVERALLS_PARALLEL: true
+
+  pip:
```

Comment:
change job name
```suggestion
  coverage:
```

Reply:
Both the conda and pip based workflows run coverage and we already have a coverage stage to finish the coveralls upload. I'm open for a better name when working with pip outside an environment.

Reply:
that's fine then

Reply:
makes sense